js/app/modules/arrangement/accordion.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 Endless Mobile, Inc.
+
+/* exported Accordion */
+
+const Gtk = imports.gi.Gtk;
+
+const Arrangement = imports.app.interfaces.arrangement;
+const Module = imports.app.interfaces.module;
+
+const TRANSITION_DURATION = 500;
+
+/**
+ * Class: Accordion
+ * Arrangement which puts its cards into an accordion-like widget.
+ * Each segment of the accordion has a title, which is always visible,
+ * and a revealer, in which lives the actual card. Clicking on the title
+ * expands the corresponding revealer.
+ */
+const Accordion = new Module.Class({
+    Name: 'Arrangement.Accordion',
+    Extends: Gtk.Grid,
+    Implements: [Arrangement.Arrangement],
+
+    _init: function (props={}) {
+        props.orientation = Gtk.Orientation.VERTICAL;
+        props.valign = Gtk.Align.START;
+        this.parent(props);
+    },
+
+    // Arrangement override
+    pack_card: function (card) {
+        let label = new Gtk.Button({
```

Comment:
yea that would be nice. Or even just being able to create a `TextCard` module directly here, without it being defined in a slot. That still doesn't really work well with current module factory setup